src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -321,6 +367,39 @@ read_port(const std::string& port_name, void* value, size_t bytes)
 
 } // namespace xrt
 
+////////////////////////////////////////////////////////////////
+// xrt_aie_event C++ API implmentations (xrt_aie.h)
+////////////////////////////////////////////////////////////////
+namespace xrt { namespace aie {
+
+event::
+event(const xrt::device& device)
+  : impl(std::move(create_event(device)))
+{}
+
```

Comment:
I wonder if this class class should define a destructor to clean up the resources acquired by `start()` if someone did not call `stop()`?

Reply:
Noted. Thanks for the suggestion Soren.